main.py

```diff
@@ -44,55 +37,66 @@ class MainApp(QtWidgets.QMainWindow, Ui_MainWindow):
     sig_send = pyqtSignal(str)
 
     def __init__(self, connection):
+        self.connection = connection
         self.data = RD()
-        atexit.register(self.exit_handler)
-
-        self.zoom = 20
-        self.radius = 0.1
 
+        # lock used for lat & lon fields # TODO Remove once RocketData refactored to have lock
         self.lock = threading.Lock()
-        self.lastgps = time.time()
-        self.lastMapUpdate = time.time()
-        self.latitude = None
+
+        self.latitude = None    # NOTE copies of what is already in RocketData TODO refactor to get from RocketData
```

Comment:
In the rest of my branch I was looking to change this to no longer being a part of main.py at all, and just retrieved from RocketData.

Should I avoid this change now, and just leave it as it is? I wasn't sure if it would be wise for the threadLoop to wait on the UI updater receiveData for the new lat and Lon. (And that is the only place using those it seems)

Reply:
I DEFINITELY still think there's a lot of refactoring potential. So 100% continue with your work if you're still down. Sorry for changing so much in the area though. It think you could continue with your branch by pulling the changes from master. Or you could start a new branch, should have the same effect in the end.

> In the rest of my branch I was looking to change this to no longer being a part of main.py at all, and just retrieved from RocketData.

Yeah that sounds great. Maybe then it would be easier to move the map thread into its own class too. That would really clean up main.py.

> I wasn't sure if it would be wise for the threadLoop to wait on the UI updater receiveData for the new lat and Lon. (And that is the only place using those it seems)

Having threads wait when there's no work to do is good. So it sounds fine, but maybe I'm missing something. We can talk more about this.

Reply:
Cool, I'm happy to pull from master!
>> In the rest of my branch I was looking to change this to no longer being a part of main.py at all, and just retrieved from RocketData.
> Yeah that sounds great. Maybe then it would be easier to move the map thread into its own class too. That would really clean up main.py.
Alright, I can attempt that in my branch!

>> I wasn't sure if it would be wise for the threadLoop to wait on the UI updater receiveData for the new lat and Lon. (And that is the only place using those it seems)
> Having threads wait when there's no work to do is good. So it sounds fine, but maybe I'm missing something. We can talk more about this.
Good point! We should definitely cover this in discussion, but I think conceptually it's totally fine.